### PR TITLE
Fix/syntaxthighlighter themes

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybooks/storybook/issues/6797

## What I did
I fixed issues with theming, typing & syntaxhighlighter component to make sure that the component will have a proper color & fixed the indentation & decreased the bundlesize while I was at it.

### CODE REVIEW

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx [hunk 1]
```diff
@@ -91,6 +91,31 @@ export interface SyntaxHighlighterState {
 
 type ReactSyntaxHighlighterProps = React.ComponentProps<typeof ReactSyntaxHighlighter>;
 
+const formatter = memoize(2)((code: string) => {
```
Comment (shilman) [inline lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx]: This seems pretty haphazard to me. Can you factor this into a separate file and provide test cases?

**Review by shilman — CHANGES_REQUESTED** [review]
Requested minor refactor/tests for code indentation

File: lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx (hunk 1 above)
Comment (ndelangen) [reply lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx]: good call

**Comment by ndelangen** [comment]
THank for the review @shilman, I extracted the function into it's own module and added tests

**Pull request merged**